best-practice-certificate-provisioning.md

```diff
@@ -0,0 +1,285 @@
+
+### DNS-SD TXT Records
+
+#### pri
+
+The DNS-SD advertisement MUST include a TXT record with key 'pri' and an integer value. Servers MAY additionally present a matching priority via the DNS-SD SRV record 'priority' and 'weight' as defined in [RFC 2782][RFC-2782]. The TXT record should be used in favour of the SRV priority and weight where these values differ, in order to overcome issues in the Bonjour and Avahi implementations. Values 0 to 99 correspond to an active EST Server API (zero being the highest priority). Values 100+ are reserved for development work to avoid colliding with a live system.
+
+#### api_label
+
+The DNS-SD advertisement MAY include a TXT record with key 'api_label' and a string. The 'api_label' key defines an arbitrary label that if present MUST be appended to the well-know EST path, for example `https://www.example.com/.well-known/est/<arbitrary label>/`. The arbitrary label allows multiple EST Server instance on a single host to be used.
+
+## EST Server Behaviour
+
+The EST server must be implemented in accordance with [RFC 7030][RFC-7030].
+
+### EST Server API
+
+The EST Server SHALL present an instance of the EST API as defined in [RFC 7030][RFC-7030].
+
+This MUST include the following API endpoints:
+
+| Operation                       | Operation path  |
+| ------------------------------- | --------------- |
+| Distribution of CA Certificates | /cacerts        |
+| Enrollment of Clients           | /simpleenroll   |
+| Re-enrollment of Clients        | /simplereenroll |
+
+
+### EST Server Authentication
+
+The EST Server MUST present a valid TLS Server Certificate signed by the CA for realm from which it is issuing certificates.
+
+### EST Server - Client Authentication
+
+The EST Server MUST authenticate the EST Client that is requesting a TLS Certificate manually or automatically using a TLS Client Certificate.
+
+The EST Server MUST support using a TLS Client Certificate, presented during the TLS handshake by the EST Client to authenticate if the Client is trusted. The TLS Client Certificate can either be signed by the current CA or an externally trusted Root CA. The EST Server MUST check the validity of the EST CLients TLS Certificate before responding to its request. The EST Server MUST provide a method to load multiple trusted Root CA's, that are used to verify TLS Client Certificate.
+
+The EST Server MAY also support manually authentication of the EST Client if; no TLS Client Certificate is presented during the TLS handshake, the TLS Client Certificate is not trusted or as an extra authentication step. The exact process for manual authentication will be implementation specific, but the EST Server MUST provide enough information to the operator so they can authenticate the EST Client. During the manual authentication the EST Server MUST respond with either HTTP 202 or HTTP 503 and the response must include a `Retry-After` header.
```

Comment:
Is the response defined in the RFC? If so a link to the section would be helpful.

Reply:
Added